### PR TITLE
Add integration tests and regtest support

### LINKED ISSUE #127: Integration test
We need an integration test, at least, for the happy path as:
- start up server
- init 2 clients
- onboard from client 1
- send from client 1 to client 2
- redeem collaboratively from client 1
- redeem unilaterally from client 2

### PR DESCRIPTION
**Pull request by louisinger**
_This PR adds e2e tests using the CLI, the ark daemon & a running ocean wallet_

TL;DR
* [CLI] add hidden & optional flag `--password` wherever it is necessary. If set, skip any user input.
* [Server] support rounds with empty shared output (no vtxos created).
* [Server] add `e2e` package running the `docker-compose.regtest.yaml` file to test CLI commands.
* Support `regtest` network.
* [CLI] add optional `--explorer` flag to `init` command. 
* run `make integrationtest` in Github CI.

it closes #127 

@altafan please review

### CODE REVIEW

File: .github/workflows/ark.integration.yaml [hunk 1]
```diff
@@ -6,6 +6,11 @@ on:
       - master
     paths:
       - "server/**"
+  pull_request:
+    branches: 
+      - master
+    paths:
+      - "server/**"
```
Comment (altafan) [inline .github/workflows/ark.integration.yaml]: nope we don't want to run the integration tests in every pr, it may take too long. Rather, we want to run them only when commits are pushed on master, so please revert this change

File: server/test/e2e/e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+	if err != nil {
+		fmt.Printf("error creating ocean wallet: %s", err)
+		os.Exit(1)
+	}
+
+	_, err = runOceanCommand("wallet", "unlock", "--password", password)
+	if err != nil {
+		fmt.Printf("error unlocking ocean wallet: %s", err)
+		os.Exit(1)
+	}
```
Comment (altafan) [inline server/test/e2e/e2e_test.go:38]: not needed, we can setup OCEAN_MNEMONIC and OCEAN_PASSWORD env vars

File: server/test/e2e/e2e_test.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+	if err != nil {
+		fmt.Printf("error funding ocean account: %s", err)
+		os.Exit(1)
+	}
+
+	_, err = runCommand("docker", "restart", "arkd")
+	if err != nil {
+		fmt.Printf("error restarting arkd: %s", err)
+		os.Exit(1)
+	}
```
Comment (altafan) [inline server/test/e2e/e2e_test.go]: why restarting the asp?

File: server/test/e2e/e2e_test.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+	time.Sleep(2 * time.Second)
+
+	_, err = runArkCommand("init", "--ark-url", "localhost:6000", "--password", password, "--network", "regtest", "--explorer", "http://host.docker.internal:3001")
```
Comment (altafan) [inline server/test/e2e/e2e_test.go]: we should be able to use docker DNS resolution:
```suggestion
	_, err = runArkCommand("init", "--ark-url", "localhost:6000", "--password", password, "--network", "regtest", "--explorer", "nigiri-chopsticks:3001")
```

File: server/test/e2e/e2e_test.go (hunk 2 above)
Comment (louisinger) [reply server/test/e2e/e2e_test.go:38]: I do think it's better to have a fresh wallet each time we run the tests (firstly to test the create accounts flow + do not mess with the utxos/txs reusing.)

File: client/common.go [hunk 5]
```diff
@@ -64,22 +72,28 @@ func verifyPassword(password []byte) error {
...
-func readPassword() ([]byte, error) {
+func readPassword(ctx *cli.Context) ([]byte, error) {
 	fmt.Print("unlock your wallet with password: ")
-	passwordInput, err := term.ReadPassword(int(syscall.Stdin))
-	fmt.Println() // new line
-	if err != nil {
-		return nil, err
+	password := []byte(ctx.String("password"))
+
+	if len(password) == 0 {
```
Comment (altafan) [inline client/common.go:78]: Let's move this check at the very beginning of the function so that we don't see the output message unlock your wallet with password: in case the --password flag is used

File: client/init.go [hunk 6]
```diff
@@ -14,7 +14,7 @@ import (
 )
 
 var (
-	passwordFlag = cli.StringFlag{
+	passwordSetFlag = cli.StringFlag{
```
Comment (altafan) [inline client/init.go]: why can't we just use a single `passwordFlag` both to encrypt/decrypt the wallet? we could just change `Usage` a little bit like: `password to encrypt/decrypt the wallet`

File: client/init.go [hunk 7]
```diff
@@ -34,34 +34,48 @@ var (
...
-	Flags:  []cli.Flag{&passwordFlag, &privateKeyFlag, &networkFlag, &urlFlag},
+	Flags:  []cli.Flag{&passwordSetFlag, &privateKeyFlag, &networkFlag, &urlFlag, &explorerFlag},
 }
 
 func initAction(ctx *cli.Context) error {
 	key := ctx.String("prvkey")
 	password := ctx.String("password")
 	net := strings.ToLower(ctx.String("network"))
 	url := ctx.String("ark-url")
+	explorer := ctx.String("explorer")
```
Comment (altafan) [inline client/init.go:51]: We need this flag for the e2e tests on ci?

File: server/internal/app-config/config.go [hunk 8]
```diff
@@ -239,11 +239,16 @@ func (c *Config) appService() error {
...
-		net = network.Testnet
+	switch c.Network.Name {
+	case "liquid":
+		return network.Liquid
+	case "testnet":
+		return network.Testnet
+	case "regtest":
+		return network.Regtest
+	default:
+		return network.Liquid
```
Comment (altafan) [inline server/internal/app-config/config.go]: nit: to avoid code repetition in switch we can write:
```suggestion
	switch c.Network.Name {
	case "testnet":
		return network.Testnet
	case "regtest":
		return network.Regtest
	case "liquid":
		fallthrough
	default:
		return network.Liquid
```

File: client/init.go (hunk 6 above)
Comment (louisinger) [reply client/init.go]: The idea of PasswordFlag was to be "hidden"

File: client/init.go (hunk 7 above)
Comment (louisinger) [reply client/init.go:51]: yes + it allows the user to use a custom endpoint

File: client/init.go (hunk 6 above)
Comment (altafan) [reply client/init.go]: I liked the idea, is it bad to reuse it also for the `init` cmd?

Comment (louisinger) [reply client/init.go]: I would say yes cause in the init the flag is required.
OR we can use an input in init if the flag is not set ?

File: client/init.go [hunk 9]
```diff
@@ -68,6 +61,10 @@ func initAction(ctx *cli.Context) error {
 
 	if len(explorer) > 0 {
 		explorerURL = explorer
+		_, network := networkFromString(net)
+		if err := testEsploraEndpoint(network, explorerURL); err != nil {
+			return err
```
Comment (altafan) [inline client/init.go]:
```suggestion
			return fmt.Errof("failed to connect with explorer: %s", err)
```
